core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -361,37 +350,28 @@ private[storage] class BlockManagerDecommissioner(
       stopped = true
     }
     try {
-      rddBlockMigrationExecutor.shutdown()
-    } catch {
-      case e: Exception =>
-        logError(s"Error during shutdown", e)
-    }
-    try {
-      shuffleBlockMigrationRefreshExecutor.shutdown()
-    } catch {
-      case e: Exception =>
-        logError(s"Error during shutdown", e)
-    }
-    try {
-      stopOffloadingShuffleBlocks()
+      rddBlockMigrationExecutor.foreach(_.shutdownNow())
     } catch {
-      case e: Exception =>
-        logError(s"Error during shutdown", e)
+      case NonFatal(e) =>
+        logError(s"Error during shutdown RDD block migration thread", e)
     }
-    logInfo("Forcing block migrations threads to stop")
-    try {
-      rddBlockMigrationExecutor.shutdownNow()
-    } catch {
-      case e: Exception =>
-        logError(s"Error during shutdown", e)
-    }
-    try {
-      shuffleBlockMigrationRefreshExecutor.shutdownNow()
-    } catch {
-      case e: Exception =>
-        logError(s"Error during shutdown", e)
+    shuffleBlockMigrationRefreshExecutor.foreach { executor =>
+      try {
+        executor.shutdownNow()
+      } catch {
+        case NonFatal(e) =>
+          logError(s"Error during shutdown shuffle block refreshing thread", e)
+      }
+      try {
+        // invoke inside `foreach` so we can avoid initiating
+        // `shuffleMigrationPool` when it's unnecessary
+        stopMigratingShuffleBlocks()
```

Comment:
What about if you just check if shuffle block migrations are configured?

Reply:
Sounds good to me!